src/cascadia/TerminalSettingsModel/CascadiaSettingsSerialization.cpp

```diff
@@ -931,43 +1074,57 @@ void SettingsLoader::_addOrMergeUserColorScheme(const winrt::com_ptr<implementat
                 userSettings.colorSchemeRemappings.emplace(newScheme->Name(), newName);
                 // And re-add it to the end.
                 userSettings.colorSchemes.emplace(newName, std::move(existingScheme));
+                return true;
             }
         }
+        return false;
     }
+    return true;
 }
 
 // As the name implies it executes a generator.
 // Generated profiles are added to .inboxSettings. Used by GenerateProfiles().
-void SettingsLoader::_executeGenerator(const IDynamicProfileGenerator& generator)
+void SettingsLoader::_executeGenerator(const IDynamicProfileGenerator& generator, std::vector<winrt::com_ptr<implementation::Profile>>& profilesList)
 {
     const auto generatorNamespace = generator.GetNamespace();
-    if (_ignoredNamespaces.contains(generatorNamespace))
-    {
-        return;
-    }
-
-    const auto previousSize = inboxSettings.profiles.size();
-
+    const auto previousSize = profilesList.size();
     try
     {
-        generator.GenerateProfiles(inboxSettings.profiles);
+        generator.GenerateProfiles(profilesList);
     }
     CATCH_LOG_MSG("Dynamic Profile Namespace: \"%.*s\"", gsl::narrow<int>(generatorNamespace.size()), generatorNamespace.data())
 
     // If the generator produced some profiles we're going to give them default attributes.
     // By setting the Origin/Source/etc. here, we deduplicate some code and ensure they aren't missing accidentally.
-    if (inboxSettings.profiles.size() > previousSize)
+    if (profilesList.size() > previousSize)
     {
         const winrt::hstring source{ generatorNamespace };
 
-        for (const auto& profile : std::span(inboxSettings.profiles).subspan(previousSize))
+        for (const auto& profile : std::span(profilesList).subspan(previousSize))
         {
             profile->Origin(OriginTag::Generated);
             profile->Source(source);
         }
     }
 }
 
+winrt::com_ptr<ExtensionPackage> SettingsLoader::_registerFragment(const winrt::Microsoft::Terminal::Settings::Model::FragmentSettings& fragment, FragmentScope scope)
+{
+    const auto src = fragment.Source();
+    if (auto extPkg = extensionPackageMap[src])

```

Comment:
generally avoid using `[]` on maps, as that will construct new items opportunistically.

I would restructure this around `find` and `emplace`.

ergo,

```c++
auto found = extensionPackageMap.find(src);
decltype(found.second) extPkg{ nullptr };
if (found != extensionPackageMap.end()) {
    extPkg = found.second;
} else {
    auto em = extensionPackageMap.emplace(src, winrt::make_self<ExtensionPackage>(src, scope));
    extPkg = em.second;
}

extPkg->Fragments().Append(fragment);
```

this consolidates the actual logic around _adding the fragment_ into a common single line. The next person to change this code doesn't need to copy their new logic on line 1117 AND 1123.